state_transfer/transfer/metadata.go

```diff
@@ -0,0 +1,12 @@
+package transfer
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ResourceMetadata defines any metadata used to create intermediary resources for state transfer
+type ResourceMetadata struct {
```

Comment:
is this something that will be used by endpoint and transport? wonder if we can pull this into a separate package?